elfpy/eth/transactions.py

```diff
@@ -66,6 +66,56 @@ def smart_contract_read(contract: Contract, function_name: str, *fn_args, **fn_k
     return {f"value{idx}": value for idx, value in enumerate(return_values)}
 
 
+def smart_contract_preview_transaction(
+    contract: Contract, signer: EthAccount, function_name: str, *fn_args
+) -> dict[str, Any]:
+    """Return from a smart contract read call
+
+    Arguments
+    ---------
+    contract : web3.contract.contract.Contract
+        The contract that we are reading from.
+    function_name : str
+        The name of the function
+    *fn_args : Unknown
+        The arguments passed to the contract method.
+    **fn_kwargs : Unknown
+        The keyword arguments passed to the contract method.
+
+    Returns
+    -------
+    dict[str, Any]
+        A dictionary of value names
+    .. todo::
+        Add better typing to the return value
+        function to recursively find component names & types
+        function to dynamically assign types to output variables
+            would be cool if this also put stuff into FixedPoint
+    """
+    # get the callable contract function from function_name & call it
+    function: ContractFunction = contract.get_function_by_name(function_name)(*fn_args)  # , **fn_kwargs)
```

Comment:
probably should do the same as in smart_contract_transact yeah?
```
        if "(" in function_name_or_signature:
            func_handle = contract.get_function_by_signature(function_name_or_signature)(*fn_args)
        else:
            func_handle = contract.get_function_by_name(function_name_or_signature)(*fn_args)
```

Reply:
yes!